alignak/objects/config.py

```diff
@@ -1629,158 +1639,114 @@ def apply_dependencies(self):
+    def set_default_values(self):
         """Fill objects properties with default value if necessary
 
         :return: None
         """
-        # Fill default for config (self)
-        super(Config, self).fill_default()
-        self.hosts.fill_default()
-        self.hostgroups.fill_default()
-        self.contacts.fill_default()
-        self.contactgroups.fill_default()
-        self.notificationways.fill_default()
-        self.checkmodulations.fill_default()
-        self.macromodulations.fill_default()
-        self.services.fill_default()
-        self.servicegroups.fill_default()
-        self.resultmodulations.fill_default()
-        self.businessimpactmodulations.fill_default()
-        self.hostsextinfo.fill_default()
-        self.servicesextinfo.fill_default()
-
-        # Now escalations
-        self.escalations.fill_default()
-
-        # Also fill default of host/servicedep objects
-        self.servicedependencies.fill_default()
-        self.hostdependencies.fill_default()
-
-        # first we create missing sat, so no other sat will
+        # First create missing satellites, so that no other satellites will
         # be created after this point
         self.fill_default_satellites()
-        # now we have all elements, we can create a default
-        # realm if need and it will be tagged to sat that do
-        # not have an realm
+        # now we have all elements, so we can create a default
+        # realm if needed and it will be tagged to satellites that do
+        # not have a realm
         self.fill_default_realm()
-        self.realms.fill_default()  # also put default inside the realms themselves
-        self.reactionners.fill_default()
-        self.pollers.fill_default()
-        self.brokers.fill_default()
-        self.receivers.fill_default()
-        self.schedulers.fill_default()
 
-        # The arbiters are already done.
-        # self.arbiters.fill_default()
+        for _, _, prop, _ in self.types_creations.values():
+            if getattr(self, prop, None) is not None:
+                # Fill the missing properties with their default values
+                getattr(self, prop).fill_default()
 
-        # Now fill some fields we can predict (like address for hosts)
-        self.fill_predictive_missing_parameters()
-
-    def fill_predictive_missing_parameters(self):
-        """Wrapper for calling fill_predictive_missing_parameters method of hosts attribute
-        Here is a special functions to fill some special
-        properties that are not filled and should be like
-        address for host (if not set, put host_name)
-
-        :return: None
-        """
-        self.hosts.fill_predictive_missing_parameters()
-        self.services.fill_predictive_missing_parameters()
+                # Fill the predictable missing properties (host_name for hosts, initial_state, ...)
+                if hasattr(getattr(self, prop), 'fill_predictable_missing_parameters'):
```

Comment:
Yes sure. But why making some specific stuff with objects ... I 'd rather use genericity that specific code. It is my way of doing ...